account-kit/react/src/components/auth/card/eoa.tsx

```diff
@@ -51,25 +61,49 @@ export const EoaConnectCard = () => {
       error={authStep.error}
       secondaryButton={{
         title: "Cancel",
-        onClick: () => setAuthStep({ type: "initial" }),
+        onClick: async () => {
+          // Ensure to stop all inflight requests
+          await signer?.disconnect();
+          setAuthStep({ type: "initial" });
+        },
       }}
     />
   );
 };
 
 export const WalletConnectCard = () => {
   const { setAuthStep, authStep } = useAuthContext("wallet_connect");
+  const { walletConnectParams } = useWalletConnectAuthConfig();
+  const { chain } = useChain();
+  const signer = useSigner();
+  const walletConnectConnector = walletConnectParams
+    ? walletConnect(walletConnectParams)
+    : null;
```

Comment:
we actually might want to throw if this connector is ever `null` this card should never be rendered without the params for wallet connect being present